src/Dfc.CourseDirectory.Core/DataStore/CosmosDb/QueryHandlers/GetVenuesByStatusHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Models;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Documents.Linq;
+
+namespace Dfc.CourseDirectory.Core.DataStore.CosmosDb.QueryHandlers
+{
+    public class GetVenuesByStatusHandler :
+        ICosmosDbQueryHandler<GetVenuesByStatus, IReadOnlyCollection<Venue>>
+    {
+        public Task<IReadOnlyCollection<Venue>> Execute(
+            DocumentClient client,
+            Configuration configuration,
+            GetVenuesByStatus request)
+        {
+            var collectionLink = UriFactory.CreateDocumentCollectionUri(
+                configuration.DatabaseId,
+                configuration.VenuesCollectionName);
+
+            var feedOptions = new FeedOptions { EnableCrossPartitionQuery = true, MaxItemCount = -1 };
+
+            return client.CreateDocumentQuery<Venue>(collectionLink, feedOptions)
+                .Where(f => f.Status == (int)request.Status)
+                .AsDocumentQuery()
+                .FetchAll();
```

Comment:
This is potentially a large query, does it need to be chunked?

Reply:
Hmmmm.

Provider with most venues has 2655 venues (including non-live), next down has 995.

It didn't fail when run across the whole of dev, so I'm inclined to leave it and see what happens, what do you think?

----
Analysis one-liner:
```
$ cat 2020-11-19_0307-venues-backup.json| jq '.' | ag UKPRN | sort | uniq -c | sort -n | tail
```

Reply:
Is it even used?

Reply:
ah, good spot, no it's not, it's a hangover for from an earlier attempt to make a cache, deleting...